## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 471049`
Commit: [`4247ebc927a65a1e577126bb21220298563be4bb`](https://github.com/NixOS/nixpkgs/commit/4247ebc927a65a1e577126bb21220298563be4bb) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/4247ebc927a65a1e577126bb21220298563be4bb..pull/471049/head))
Merge: [`e1f976db8343bff3d9a6b0618ea8fb8d14d4dcfa`](https://github.com/NixOS/nixpkgs/commit/e1f976db8343bff3d9a6b0618ea8fb8d14d4dcfa)

Logs: https://github.com/daspk04/nixpkgs-review-gha/actions/runs/20255914041


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.fastcan</li>
    <li>python312Packages.fastcan.dist</li>
    <li>python313Packages.fastcan</li>
    <li>python313Packages.fastcan.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.fastcan</li>
    <li>python312Packages.fastcan.dist</li>
    <li>python313Packages.fastcan</li>
    <li>python313Packages.fastcan.dist</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.fastcan</li>
    <li>python312Packages.fastcan.dist</li>
    <li>python313Packages.fastcan</li>
    <li>python313Packages.fastcan.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python312Packages.fastcan</summary>
<pre>                   ^^^^^^^^^^^
/nix/store/978h7w9sbz3f6qi6ra4pp6prqflgqyqq-python3-3.12.12/lib/python3.12/pathlib.py:840: in stat
    return os.stat(self, follow_symlinks=follow_symlinks)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
=============================== warnings summary ===============================
../../../store/10qbm884hnign19j4zfaxwiyw1pf16kj-python3.12-pytest-8.4.2/lib/python3.12/site-packages/_pytest/cacheprovider.py:475
  /nix/store/10qbm884hnign19j4zfaxwiyw1pf16kj-python3.12-pytest-8.4.2/lib/python3.12/site-packages/_pytest/cacheprovider.py:475: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/nodeids: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-yizk1sbc&#x27;
    config.cache.set(&quot;cache/nodeids&quot;, sorted(self.cached_nodeids))

../../../store/10qbm884hnign19j4zfaxwiyw1pf16kj-python3.12-pytest-8.4.2/lib/python3.12/site-packages/_pytest/cacheprovider.py:429
  /nix/store/10qbm884hnign19j4zfaxwiyw1pf16kj-python3.12-pytest-8.4.2/lib/python3.12/site-packages/_pytest/cacheprovider.py:429: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/lastfailed: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-tkfpijov&#x27;
    config.cache.set(&quot;cache/lastfailed&quot;, self.lastfailed)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR ../../.. - PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
========================= 2 warnings, 1 error in 0.19s =========================
ERROR: found no collectors for /nix/store/qv49bwmvlv02mj7rajr36x08rkfbdq9j-python3.12-fastcan-0.5.0/tests
</pre>
</details>
<details>
<summary>python313Packages.fastcan</summary>
<pre>                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9/lib/python3.13/pathlib/_local.py:515: in stat
    return os.stat(self, follow_symlinks=follow_symlinks)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
=============================== warnings summary ===============================
../../../store/1c4312acvqqam02x5xmfzxqi80xsk1xx-python3.13-pytest-8.4.2/lib/python3.13/site-packages/_pytest/cacheprovider.py:475
  /nix/store/1c4312acvqqam02x5xmfzxqi80xsk1xx-python3.13-pytest-8.4.2/lib/python3.13/site-packages/_pytest/cacheprovider.py:475: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/nodeids: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-x03vpmur&#x27;
    config.cache.set(&quot;cache/nodeids&quot;, sorted(self.cached_nodeids))

../../../store/1c4312acvqqam02x5xmfzxqi80xsk1xx-python3.13-pytest-8.4.2/lib/python3.13/site-packages/_pytest/cacheprovider.py:429
  /nix/store/1c4312acvqqam02x5xmfzxqi80xsk1xx-python3.13-pytest-8.4.2/lib/python3.13/site-packages/_pytest/cacheprovider.py:429: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/lastfailed: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-l0agaf2c&#x27;
    config.cache.set(&quot;cache/lastfailed&quot;, self.lastfailed)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR ../../.. - PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
========================= 2 warnings, 1 error in 0.17s =========================
ERROR: found no collectors for /nix/store/bx8v2zplqyvr9gsp91viyv9fby04j38l-python3.13-fastcan-0.5.0/tests
</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.fastcan</li>
    <li>python312Packages.fastcan.dist</li>
    <li>python313Packages.fastcan</li>
    <li>python313Packages.fastcan.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.fastcan</summary>
<pre>                   ^^^^^^^^^^^
/nix/store/nrk1773m026hlgkk7vlpvc2s6zlwf1rh-python3-3.12.12/lib/python3.12/pathlib.py:840: in stat
    return os.stat(self, follow_symlinks=follow_symlinks)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
=============================== warnings summary ===============================
../../../store/bdghf5p7l2mnv7wx6l6mz5x65mla7gvl-python3.12-pytest-8.4.2/lib/python3.12/site-packages/_pytest/cacheprovider.py:475
  /nix/store/bdghf5p7l2mnv7wx6l6mz5x65mla7gvl-python3.12-pytest-8.4.2/lib/python3.12/site-packages/_pytest/cacheprovider.py:475: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/nodeids: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-hjkp58ev&#x27;
    config.cache.set(&quot;cache/nodeids&quot;, sorted(self.cached_nodeids))

../../../store/bdghf5p7l2mnv7wx6l6mz5x65mla7gvl-python3.12-pytest-8.4.2/lib/python3.12/site-packages/_pytest/cacheprovider.py:429
  /nix/store/bdghf5p7l2mnv7wx6l6mz5x65mla7gvl-python3.12-pytest-8.4.2/lib/python3.12/site-packages/_pytest/cacheprovider.py:429: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/lastfailed: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-ubkeunbb&#x27;
    config.cache.set(&quot;cache/lastfailed&quot;, self.lastfailed)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR ../../.. - PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
========================= 2 warnings, 1 error in 0.09s =========================
ERROR: found no collectors for /nix/store/hs0wvnqpzpickq2aa85z1k2339jn885j-python3.12-fastcan-0.5.0/tests
</pre>
</details>
<details>
<summary>python313Packages.fastcan</summary>
<pre>                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/nix/store/xcjk9ill54kjk8mzgq6yydnx9015lidg-python3-3.13.9/lib/python3.13/pathlib/_local.py:515: in stat
    return os.stat(self, follow_symlinks=follow_symlinks)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
=============================== warnings summary ===============================
../../../store/m5hrx2ah4vfvys1vr56gafcr7mp00z5j-python3.13-pytest-8.4.2/lib/python3.13/site-packages/_pytest/cacheprovider.py:475
  /nix/store/m5hrx2ah4vfvys1vr56gafcr7mp00z5j-python3.13-pytest-8.4.2/lib/python3.13/site-packages/_pytest/cacheprovider.py:475: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/nodeids: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-ofscx1f5&#x27;
    config.cache.set(&quot;cache/nodeids&quot;, sorted(self.cached_nodeids))

../../../store/m5hrx2ah4vfvys1vr56gafcr7mp00z5j-python3.13-pytest-8.4.2/lib/python3.13/site-packages/_pytest/cacheprovider.py:429
  /nix/store/m5hrx2ah4vfvys1vr56gafcr7mp00z5j-python3.13-pytest-8.4.2/lib/python3.13/site-packages/_pytest/cacheprovider.py:429: PytestCacheWarning: could not create cache path /nix/.pytest_cache/v/cache/lastfailed: [Errno 1] Operation not permitted: &#x27;/nix/pytest-cache-files-yl7kzkim&#x27;
    config.cache.set(&quot;cache/lastfailed&quot;, self.lastfailed)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR ../../.. - PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/.Trashes&#x27;
========================= 2 warnings, 1 error in 0.06s =========================
ERROR: found no collectors for /nix/store/sj0jwp7gsyix3rpm3mkml9wbg6yd8cc9-python3.13-fastcan-0.5.0/tests
</pre>
</details>
</details>

--- a/pkgs/development/python-modules/fastcan/default.nix
+++ b/pkgs/development/python-modules/fastcan/default.nix
@@ -1,0 +1,64 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  cython,
+  meson-python,
+  numpy,
+  setuptools,
+
+  # nativeBuildInputs
+  pkg-config,
+
+  # dependencies
+  scikit-learn,
+
+  # tests
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "fastcan";
+  version = "0.5.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "scikit-learn-contrib";
+    repo = "fastcan";
+    tag = "v${version}";
+    hash = "sha256-1ncdzBMJYEwTkpLXS64g+SaEbsiYslX7zN4xbGjUsAA=";
+  };
+
+  build-system = [
+    cython
+    meson-python
+    numpy
+    setuptools
+  ];
+
+  nativeBuildInputs = [ pkg-config ];
+
+  dependencies = [ scikit-learn ];
+
+  # Prevent pytest from importing the required python modules from the source instead of $out
+  preCheck = ''
+    rm -f fastcan/__init__.py
+    echo "" > fastcan/narx/__init__.py
+  '';
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "fastcan" ];
+
+  meta = {
+    description = "Library for fast canonical-correlation-based search algorithm";
+    homepage = "https://github.com/scikit-learn-contrib/fastcan";
+    changelog = "https://github.com/scikit-learn-contrib/fastcan/releases/tag/${src.tag}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ daspk04 ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -5208,6 +5208,8 @@ self: super: with self; {
 
   fastcache = callPackage ../development/python-modules/fastcache { };
 
+  fastcan = callPackage ../development/python-modules/fastcan { };
+
   fastcore = callPackage ../development/python-modules/fastcore { };
 
   fastcrc = callPackage ../development/python-modules/fastcrc { };

